src/sentry/http.py

```diff
@@ -151,7 +151,9 @@ def fetch_file(
         domain_key = "source:blacklist:v2:{}".format(md5_text(domain).hexdigest())
         domain_result = cache.get(domain_key)
         if domain_result:
+            domain_result = dict(domain_result)
             domain_result["url"] = url
+            domain_result["domain_blacklisted"] = True
```

Comment:
You've only added to the cache path. There is another branch `if domain_lock_enabled` below after which the original error is raised. 

Is the blacklist only written in this function or also manually? It looks like the errors are not just blacklists but really all sorts of errors that happen during fetching. If so, rather than adding in the `domain_lock_enabled` branch, you could add this just for the errors that constitute an actual blacklist.